cinn/auto_schedule/search_space/auto_gen_rule/add_cache_read.cc

```diff
@@ -0,0 +1,187 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cinn/auto_schedule/search_space/auto_gen_rule/add_cache_read.h"
+
+#include <glog/logging.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "cinn/auto_schedule/analysis/analyze_ir.h"
+#include "cinn/auto_schedule/search_space/auto_gen_rule/auto_gen_rule.h"
+#include "cinn/auto_schedule/search_space/auto_gen_rule/multi_level_tiling.h"
+#include "cinn/common/target.h"
+#include "cinn/ir/buffer.h"
+#include "cinn/ir/collect_ir_nodes.h"
+#include "cinn/ir/ir.h"
+#include "cinn/ir/ir_base.h"
+#include "cinn/ir/ir_printer.h"
+#include "cinn/ir/ir_schedule.h"
+#include "cinn/ir/tensor.h"
+#include "cinn/optim/ir_copy.h"
+
+namespace cinn {
+namespace auto_schedule {
+
+RuleApplyType AddCacheRead::Init(ir::IRSchedule* ir_schedule) {
+  ir_schedule_        = ir_schedule;
+  auto block_realizes = ir_schedule_->GetAllBlocks();
+  applicable_schedule_blocks_.clear();
+  num_applicable_ = 0;
+  for (size_t i = 0; i < block_realizes.size(); ++i) {
+    ir::ScheduleBlockRealize* sch_block_realize = block_realizes[i].As<ir::ScheduleBlockRealize>();
+    AnalyzeScheduleBlockReadWriteBuffer(sch_block_realize->schedule_block.As<ir::ScheduleBlock>());
+    if (MeetCondition(block_realizes[i])) {
+      ++num_applicable_;
+      applicable_schedule_blocks_.push_back(block_realizes[i]);
+    }
+  }
+  VLOG(6) << "Collect applicable_schedule_blocks_:" << num_applicable_;
+
+  cache_memory_type_ = kMemoryTypes.at(target_->arch);
+
+  return num_applicable_ > 0 ? RuleApplyType::kApplyAndSkipAllRules : RuleApplyType::kCannotApply;
+}
+
+void AddCacheRead::Apply(int index) {
+  ir::Expr sch_block_expr                     = applicable_schedule_blocks_[index];
+  ir::ScheduleBlockRealize* sch_block_realize = sch_block_expr.As<ir::ScheduleBlockRealize>();
+  ir::ScheduleBlock* sch_block                = sch_block_realize->schedule_block.As<ir::ScheduleBlock>();
+  std::string block_name                      = sch_block->name;
+
+  // Analyze which buffers can be cached
+  std::vector<std::string> read_tensor_strs;
+  ir::CollectIRNodesWithoutTensor(sch_block->body, [&](const Expr* x) {
+    if (x->As<ir::Load>()) read_tensor_strs.push_back(x->As<ir::Load>()->tensor.as_tensor()->name);
+    return x->As<ir::Load>() != nullptr;
+  });
+  for (int i = 0; i < read_tensor_strs.size(); ++i) {
+    VLOG(6) << "read tensors[" << i << "]: " << read_tensor_strs[i];
+  }
+
+  std::unordered_set<std::string> write_tensor_strs;
+  ir::CollectIRNodesWithoutTensor(sch_block->body, [&](const Expr* x) {
+    if (x->As<ir::Store>()) write_tensor_strs.insert(x->As<ir::Store>()->tensor.as_tensor()->name);
+    return x->As<ir::Store>() != nullptr;
+  });
+  for (const auto& write_tensor_str : write_tensor_strs) {
+    VLOG(6) << "write tensors: " << write_tensor_str;
+  }
```

Comment:
```suggestion
if (VLOG_IS_ON(6)) {
  for (const auto& write_tensor_str : write_tensor_strs) {
    VLOG(6) << "write tensors: " << write_tensor_str;
  }
}
```
对于可能影响性能的debug日志，可以添加VLOG_IS_ON进行屏蔽

Reply:
done